src/main/java/uk/gov/justice/hmpps/prison/api/resource/ScheduleResource.java

```diff
@@ -117,12 +119,13 @@ public List<PrisonerSchedule> getActivitiesAtAllLocationsByDateRange(@PathVariab
             @ApiResponse(responseCode = "404", description = "Requested resource not found.", content = {@Content(mediaType = "application/json", schema = @Schema(implementation = ErrorResponse.class))}),
             @ApiResponse(responseCode = "500", description = "Unrecoverable error occurred whilst processing request.", content = {@Content(mediaType = "application/json", schema = @Schema(implementation = ErrorResponse.class))})})
     @Operation(summary = "Get all Prisoner activities for given date.", description = "Get count of suspended prisoner activities for given date range")
-    @GetMapping("/{agencyId}/count-activities")
+    @PostMapping("/{agencyId}/count-activities")
     public PrisonerActivitiesCount getCountActivitiesByDateRange(@PathVariable("agencyId") @Parameter(description = "The prison.", required = true, example = "MDI") final String agencyId,
                                                                  @RequestParam(value = "fromDate") @DateTimeFormat(iso = ISO.DATE) @Parameter(description = "From date of whereabouts list") @NotNull final LocalDate fromDate,
                                                                  @RequestParam(value = "toDate") @DateTimeFormat(iso = ISO.DATE) @Parameter(description = "To Date of whereabouts list") @NotNull final LocalDate toDate,
-                                                                 @RequestParam(value = "timeSlots") @Parameter(description = "List of one or more of AM, PM or ED") @NotEmpty final List<TimeSlot> timeSlots) {
-        return schedulesService.getCountActivities(agencyId, fromDate, toDate, timeSlots);
+                                                                 @RequestParam(value = "timeSlots") @Parameter(description = "Set of one or more of AM, PM or ED") @NotEmpty final Set<TimeSlot> timeSlots,
+                                                                 @RequestBody @Parameter(description = "Map of booking IDs to their occurrence counts") final Map<Long, Long> attendanceCounts) {
```

Comment:
Changed to a `POST` so can send in a map of the booking ids (and their counts) that have been recorded in whereabouts.